doc/DESI_SPECTRO_DATA/NIGHT/EXPID/desi-EXPID.rst

```diff
@@ -16,138 +16,476 @@ There is one HDU per spectrograph camera with EXTNAMEs like
 B0, B1, ... R0, R1, ... Z8, Z9.  The structure of each of these is
 the same; only one is explicitly documented below.
 
-====== ======= ===== ===================
-Number EXTNAME Type  Contents
-====== ======= ===== ===================
-HDU0_  PRIMARY IMAGE Blank except for header keywords
-HDU1_  B0      IMAGE Raw data from the b0 spectrograph
-====== ======= ===== ===================
+================= ========= ======== ====================================
+Number            EXTNAME   Type     Contents
+================= ========= ======== ====================================
+HDU0_             SPS       IMAGE    Blank except for header keywords
+HDU1_             z0        IMAGE    Raw data from the b0 spectrograph
+`HDU2 -- HDU6`_   various   IMAGE    Raw data similar to b0 spectrograph.
+HDU7_             SPECTCONS BINTABLE Telemetry data
```

Comment:
A full desi*.fits.fz file will have HDU 0 = SPS (guaranteed), HDU 1-30 as B0,R0,Z0, ... B9,R9,Z9 but in arbitrary order and individual cameras could be validly missing, plus a EXTNAME=SPECTCONS HDU which I think will always be last, but perhaps not.  Let's document the full set that could be there and document that individual cameras can be missing (i.e. code shouldn't assume that a given camera is always present in every exposure -- that depends upon the state of the hardware and the ICS configuration).  For the moment I'm not making these updates myself because I'm not sure which portions need to be specifically formatted for the validator to work, vs. are flexible for humans to read.

Minor mismatch between EXTNAME=z0 vs. Contents comment b0.

Otherwise looks good for this round of updates.

Reply:
The *current* code needs to know the exact number of HDUs in order to function, so I temporarily altered the number of HDUs to match the `20190306/00001209/desi-00001209.fits.fz` so I could run tests.  I will switch that back now.